src/MinterGateway.sol

```diff
@@ -602,10 +602,70 @@ contract MinterGateway is IMinterGateway, ContinuousIndexing, ERC712Extended {
     }
 
     /// @inheritdoc IMinterGateway
-    function getPenaltyForMissedCollateralUpdates(address minter_) external view returns (uint240) {
-        uint112 penaltyPrincipal_ = _getPenaltyPrincipalForMissedCollateralUpdates(minter_);
+    function getPenalties(
+        address minter_
+    ) external view returns (uint240 missedIntervalPenalty_, uint240 undercollateralizedPenalty_) {
+        uint112 principalOfActiveOwedM_ = principalOfActiveOwedMOf(minter_);
+
+        if (principalOfActiveOwedM_ == 0) return (0, 0);
+
+        uint32 penaltyRate_ = penaltyRate();
+
+        if (penaltyRate_ == 0) return (0, 0);
+
+        MinterState storage minterState_ = _minterStates[minter_];
+
+        (uint40 missedIntervals_, uint40 missedUntil_) = _getMissedCollateralUpdateParameters(
+            minterState_.updateTimestamp,
+            minterState_.penalizedUntilTimestamp,
+            updateCollateralInterval()
+        );
+
+        unchecked {
+            // As an edge case precaution, cap the penalty principal to type(uint112).max.
+            uint112 missedIntervalPenaltyPrincipal_ = UIntMath.bound112(
+                (uint256(principalOfActiveOwedM_) * missedIntervals_ * penaltyRate_) / ONE
+            );
+
+            principalOfActiveOwedM_ = UIntMath.bound112(
+                uint256(principalOfActiveOwedM_) + uint256(missedIntervalPenaltyPrincipal_)
+            );
+
+            missedIntervalPenalty_ = _getPresentAmount(missedIntervalPenaltyPrincipal_);
+        }
```

Comment:
A lot of this logic is part of `_imposePenaltyIfMissedCollateralUpdates` and `_imposePenalty`, we should try to create helper functions that would handle the calculations. Same goes for the calculation of `undercollateralizedPenalty_`.